ntbs-service/Services/NotificationService.cs

```diff
@@ -37,8 +36,7 @@ public class NotificationService : INotificationService
         public async Task UpdatePatientAsync(Notification notification, PatientDetails patient)
         {
             await UpdatePatientFlags(patient);
-            context.Attach(notification);
-            notification.PatientDetails = patient;
+            context.Entry(notification.PatientDetails).CurrentValues.SetValues(patient);

```

Comment:
What was here before led to the Audit.NET EF library logging an "Insert" statement rather than an update, which made me think just attaching the notification is probably not correct - owned properties are a bit weird.
I tried changing it to `context.Attach(notification.PatientDetails).State = EntityState.Modified`, but then every single property is marked as modified and you can't actually see what changes were made.

The current way is the one described in https://stackoverflow.com/questions/7106211/entity-framework-why-explicitly-set-entity-state-to-modified, and makes Audit.NET work correctly (i.e. only properties that got changed are marked as such). I know in general we shouldn't change our code to make something else work, but I'm pretty sure whoever wrote Audit.NET EF has a better understanding of how it tracks changes than me lol.
Also, it works and this syntax was also already used in socialriskfactors below, so I think we're good now.

Reply:
Yes, I'm also very much willing to trust that Audit.NET got it more right than us :D
The change seems absolutely fine at a glance, so happy to keep it here! Prefer the one-liner anyways ;)